main.py

```diff
@@ -1,17 +1,99 @@
+* We suggest using a convolutional neural network.
+* TensorFlow Keras has the CIFAR-10 dataset as a module, so you don't need to manually download and unpack it.
+"""
+
+# Import whatever libraries/modules you need
+import matplotlib.pyplot as plt
+import numpy as np
+
+import ssl
+
+import tensorflow as tf
+from tensorflow.keras import datasets, layers, models
+
+# ssl Verification error fix 
+ssl._create_default_https_context = ssl._create_unverified_context
+
+# Loading the data-set and normalizing the rgb values between 0 and 1
+(trainImages, trainLables), (testImages, testLables) = datasets.cifar10.load_data()
+
+# Normalize pixel values to be between 0 and 1
+trainImages, testImages = trainImages / 255.0, testImages / 255.0
+
+# Verifying the data 
+classNames = ['airplane', 'automobile', 'bird', 'cat', 'deer',
+               'dog', 'frog', 'horse', 'ship', 'truck']
+
+## Defining a function to plot the dataset(for testing)
+def plot_image(i, predictionsArray, trueLabel, img):
+    trueLabel, img = trueLabel[i], img[i]
+    plt.grid(False)
+    plt.xticks([])
+    plt.yticks([])
+
+    plt.imshow(img, cmap=plt.cm.binary)
+
+    predictedLabel = np.argmax(predictionsArray)
+    if predictedLabel == trueLabel:
+        color = 'blue'
+    else:
+        color = 'red'
+
+    plt.xlabel("{} {:2.0f}% ({})".format(classNames[predictedLabel],
+                                    100*np.max(predictionsArray),
+                                    classNames[trueLabel]),
+                                    color=color)
+
+#Creating a Convolutional Neural Network. 
+model = models.Sequential()
+
+#Defining the convulational a stack of COnv2D and MaxPolling2D layers
+model.add(layers.Conv2D(32, (3, 3), activation='relu', input_shape=(32, 32, 3)))
+model.add(layers.MaxPooling2D((2, 2)))
+model.add(layers.Conv2D(64, (3, 3), activation='relu'))
+model.add(layers.MaxPooling2D((2, 2)))
+model.add(layers.Conv2D(64, (3, 3), activation='relu'))
+
+#Displaying the model architecture
+print("Before Flattening: ")
+model.summary()
+
+model.add(layers.Flatten())
+model.add(layers.Dense(64, activation='relu'))
+model.add(layers.Dense(10))
+
+print("After Flattening: ")
+model.summary()
+
+#Compiling and training the model
+model.compile(optimizer='adam',
+              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+              metrics=['accuracy'])
+
+history = model.fit(trainImages, trainLables, epochs=10, 
+                    validation_data=(testImages, testLables))
+
+#Evaluating the model using the test dataset and measuring the accuracy
+plt.plot(history.history['accuracy'], label='accuracy')
+plt.plot(history.history['val_accuracy'], label = 'val_accuracy')
+plt.xlabel('Epoch')
+plt.ylabel('Accuracy')
+plt.ylim([0.5, 1])
+plt.legend(loc='lower right')
+
+testLoss, testAccuracy = model.evaluate(testImages,  testLables, verbose=2)
+
+print("Test Loss: ", testLoss)
+print("Test Accuracy: ", testAccuracy)
+

```

Comment:
the idea is to group most of the code into functions and to have main function calling them one by one